Gems/AtomLyIntegration/EditorModeFeedback/Assets/Shaders/EditorModeCommon.azsli

```diff
@@ -77,3 +83,60 @@ partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
         return CalculateOutputColor(inColor, finalEffect, PassSrg::CalculateFinalBlendAmount(t));
     }
 }
+
+//! Calculates the view position and normal from the fragment position.
+void ComputeViewNormalAndPosition(VSOutput IN, out float3 outPositionWS, out float3 outNormalWS)
+{
+    const float2 pixelSize = PassSrg::m_maskDimensions.zw;  // How big a pixel is in screen UV space
+    const float2 halfPixel = pixelSize * 0.5f;
+    const int2   screenPos = IN.m_position.xy;              // The coordinates of the screen pixel being shaded
+    const float2 screenUV  = IN.m_texCoord.xy;              // The UV value [0, 1] of the screen pixel
+
+    // Do 2 depth gather ops to get 5 depth values (cross centered on pixel being shaded). Reminder that gather is laid out like so:
+    //  W Z
+    //  X Y
+    float4 depthUpLeft = PassSrg::m_depth.Gather(PassSrg::PointSampler, screenUV - halfPixel);
+    float4 depthBottomRight = PassSrg::m_depth.Gather(PassSrg::PointSampler, screenUV + halfPixel);
+
+    depthUpLeft = CalculateLinearDepth(depthUpLeft);
+    depthBottomRight = CalculateLinearDepth(depthBottomRight);
+
+    float3 positionVS = ViewSrg::GetViewSpacePosition(screenUV, depthUpLeft.y);
+
+    float3 diffX;
+    {
+        float3 positionLeft  = ViewSrg::GetViewSpacePosition( float2(screenUV.x - pixelSize.x, screenUV.y), depthUpLeft.x);
+        float3 positionRight = ViewSrg::GetViewSpacePosition( float2(screenUV.x + pixelSize.x, screenUV.y), depthBottomRight.z);
+        float3 diffLeft = positionVS - positionLeft;
+        float3 diffRight = positionRight - positionVS;
+        diffX = (abs(diffLeft.z) < abs(diffRight.z)) ? diffLeft : diffRight;
+    }
+    float3 diffY;
+    {
+        float3 positionUp   = ViewSrg::GetViewSpacePosition( float2(screenUV.x, screenUV.y - pixelSize.y), depthUpLeft.z);
+        float3 positionDown = ViewSrg::GetViewSpacePosition( float2(screenUV.x, screenUV.y + pixelSize.y), depthBottomRight.x);
+        float3 diffUp = positionVS - positionUp;
+        float3 diffDown = positionDown - positionVS;
+        diffY = (abs(diffUp.z) < abs(diffDown.z)) ? diffUp : diffDown;
+    }
+
+    float3 normalVS = normalize( cross(diffX, diffY) );
+    
```

Comment:
Maybe I missed it but why cant this section of the code be refactored to a common function that can be called by Editor feedback shader as well as the shadow shader?

Reply:
The params sent by both are subtly different enough that one of the two will require clunky boilerplate prep for the params for each call. I would suggest we get these changes in and then if someone wishes to have another crack at reconciling the two implementations into one generic implementation then the door is open to do so 👍

Reply:
To be honest I would be ok with getting this in now if it was blocking a feature or a bug and create a ghi to refactor later but the thing is no one is using these functions. So we are essentially just adding un-factored code in the hopes that it may be used in future. This just adds more confusion and tech debt. My recommendation here it to just reference the shadow code directly in the blog and just mention how it can be refactored and used in these shaders if an effect needs to built in future.

Reply:
In that case, I think your second suggestion is the better approach: As the function is short enough, I will put the source code directly in the blog post and reference it there 👍

Reply:
Updated blog post with this suggestion here: https://github.com/o3de/o3de.org/pull/2389